frappe/website/serve.py

```diff
@@ -0,0 +1,276 @@
+import frappe
+import os, mimetypes
+
+from werkzeug.wrappers import Response
+from werkzeug.wsgi import wrap_file
+
+from frappe.website.render import (resolve_path, build_response)
+from frappe.website.redirect import resolve_redirect
+from frappe.website.router import get_start_folders
+from frappe.website.utils import (get_home_page, can_cache, delete_page_cache,
+	get_toc, get_next_link)
+from frappe.website.doctype.website_settings.website_settings import get_website_settings
+
+def render(path=None, http_status_code=None):
+	"""render html page"""
+	if not path:
+		path = frappe.local.request.path
+
+	try:
+		path = path.strip('/ ')
+		resolve_redirect(path)
+		path = resolve_path(path)
+		data = None
+
+		response = StaticPage(path).get()
+		if not response:
+			response = TemplatePage(path).get()
+		if not response:
+			response = DocTypePage(path).get()
+		if not response:
+			response = TemplatePage('404').get()
```

Comment:
shouldn't we find what type of page it is based on path?

Reply:
there are no rules that classify directly unless the route is resolved.